src/line-chart.js

```diff
@@ -53,7 +53,7 @@ class LineChart extends AbstractChart {
             cx={cx}
             cy={cy}
             r="4"
-            fill={(dotColor && dotColor(x, i)) || this.getColor(dataset, 0.9)}
+            fill={typeof dotColor === 'function' ? dotColor(x, i) : this.getColor(dataset, 0.9)}
```

Comment:
It may be necessary to benchmark test this with a large data set.
Using a Ternary operator versus the logical comparison can be significant in terms of performance in some cases.
This may or may not be one of them.

Reply:
While I agree that benchmarking is important, in this particular case I made this change to make the code less ambiguous, because you could potentially get the `this.getColor(...)` called even if `dotColor` is provided and is a function.

If you are interested in performance, I would like to share a talk I watched yesterday https://youtu.be/lXbjICP6V44?t=19221 I found it very interesting, and it made me question some assumptions I make when it comes to performance.

Reply:
Thank you for the link, interesting things there. 😄  (Haven't watched it all yet, but plan to)

In my experience I've had to swap out ternary ops for logical ops in nested loops such as this when working with extremely large sets of data, so while it's a micro-optimization, it's one that has served me well in the past, and which is why I had brought it up.

Not all cases are the same, and it's also possible that you will have better performance under the native runtimes versus the browser itself.